WalletWasabi.Fluent/ViewModels/Wallets/Settings/ResyncWalletViewModel.cs

```diff
@@ -0,0 +1,44 @@
+using System.Reactive.Linq;
+using ReactiveUI;
+using WalletWasabi.Fluent.Validation;
+using WalletWasabi.Fluent.ViewModels.Dialogs.Base;
+using WalletWasabi.Models;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets.Settings;
+
+[NavigationMetaData(Title = "Resync Wallet", NavigationTarget = NavigationTarget.CompactDialogScreen)]
+public partial class ResyncWalletViewModel : DialogViewModelBase<int?>
+{
+	[AutoNotify] private string _startingHeight = "";
+
+	public ResyncWalletViewModel()
+	{
+		this.ValidateProperty(x => x.StartingHeight, ValidateStartingHeight);
+
+		SetupCancel(false, true, true);
+
+		NextCommand = ReactiveCommand.Create(
+			() =>
+				Close(DialogResultKind.Normal, StartingHeight is "" ? 0 : int.Parse(StartingHeight)),
+			this.WhenAnyValue(x => x.StartingHeight).Select(_ => !Validations.Any));
+	}
+
+	private void ValidateStartingHeight(IValidationErrors errors)
```

Comment:
maybe could make it accept only numbers,
instead of displaying errors?

Reply:
When I write features like this, I always copy components. In this case, the component copied is the `MinGapLimit` Text Box in the `AdvancedRecoveryScreen`. I can check if this is easy, but if the only example in the software is the `CurrencyEntryBox`.........

Reply:
It was easy